addons/test_mail/tests/test_mail_channel.py

```diff
@@ -299,6 +299,39 @@ def test_channel_auto_unsubscribe_archived_or_deleted_users(self):
         self.assertEqual(self.test_channel.channel_partner_ids, self.user_employee.partner_id | test_partner)
         self.assertEqual(test_chat.channel_partner_ids, self.user_employee.partner_id | test_partner)
 
+    def test_channel_unfollow_should_also_unsubscribe_the_partner(self):
+        self.assertEqual(self.test_channel.message_channel_ids, self.test_channel)
+        self.assertEqual(self.test_channel.message_partner_ids, self.env['res.partner'])
+        self.assertEqual(self.test_channel.channel_partner_ids, self.env['res.partner'])
+
+        self.test_channel.message_subscribe(self.test_partner.ids)
+        self.assertEqual(self.test_channel.message_channel_ids, self.test_channel)
+        self.assertEqual(self.test_channel.message_partner_ids, self.test_partner)
+        self.assertEqual(self.test_channel.channel_partner_ids, self.env['res.partner'])
+
+        self._leave_channel(self.test_channel, self.test_partner)
+        self.assertEqual(self.test_channel.message_channel_ids, self.test_channel)
+        self.assertEqual(self.test_channel.message_partner_ids, self.env['res.partner'])
+        self.assertEqual(self.test_channel.channel_partner_ids, self.env['res.partner'])
+
+    @mute_logger('odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
+    def test_channel_unfollow_should_not_post_message_if_the_partner_has_been_removed(self):
+        self.env['ir.config_parameter'].set_param('mail.catchall.domain', 'schlouby.fr')
+        self.test_channel.write({'email_send': False})
+        self.user_employee.write({'notification_type': 'email'})
+        self.test_channel.message_subscribe(self.user_employee.partner_id.ids)
+
+        self._join_channel(self.test_channel, self.test_partner)
+        with self.mock_mail_gateway():
+            self.test_channel._action_unfollow(self.test_partner)
+        self.assertSentEmail(self.test_partner, [self.partner_employee])
+        self.assertEqual(len(self._mails), 1)
```

Comment:
Actually I don't understand this whole test. Why should we send any email if nobody is posting on it ?